From 7502db5f1cb4f7f891354383766fb44c8b4a877f Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 20 May 2021 19:51:02 +0300
Subject: [PATCH] [SW-42558] gaudi: remove disable iatu access func

The function is not needed anymore as the iATU is configured by FW and
host should not access it anymore.
Thus remove the preboot_disable_iatu_access() function and all calls.

Change-Id: I978ef9323a879b766ad0a6ffbf5648152398760b
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 3c883d1..d46715a 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -489,17 +489,6 @@
 		regmap_writel(i, mmGIC_DISTRIBUTOR__6_GICD_CLRSPI_NSR);
 }
 
-static bool iatu_disabled;
-
-void preboot_disable_iatu_access(void)
-{
-	if (iatu_disabled || !preboot_security_is_enabled())
-		return;
-
-	preboot_ack_razwi_irq();
-	iatu_disabled = true;
-}
-
 u32 dev_pci_dbi_cfg_offset_get(void)
 {
 	return (u32)mmPCIE_DBI_BASE & 0xFFFFFF;
diff --git a/include/preboot.h b/include/preboot.h
index 424ca0c..2473c0b 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -166,7 +166,6 @@
 void preboot_dev_enable_caps(void);
 void preboot_set_enabled_cap(enum hl_fw_en_caps en_cap);
 void preboot_finalize_enabled_caps(void);
-void preboot_disable_iatu_access(void);
 void preboot_ack_razwi_irq(void);
 const char *preboot_get_version(enum preboot_vers ver);
 void hl_ram_init(void);
diff --git a/preboot/module_i2c_slave.c b/preboot/module_i2c_slave.c
index 84cb35e..fc62476 100644
--- a/preboot/module_i2c_slave.c
+++ b/preboot/module_i2c_slave.c
@@ -161,9 +161,6 @@
 
 void i2cslave_fit_run_image(void)
 {
-	/* TODO: temporary fix, will be removed soon */
-	preboot_disable_iatu_access();
-
 	fit_run_image();
 }
 
diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index ab895c8..70c226c 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -46,8 +46,6 @@
 	module_lkd_enable();
 }
 
-__weak void preboot_disable_iatu_access(void) {}
-
 static void module_lkd_send_status(enum comms_sts sts,
 				   enum comms_ram_types ram_type,
 				   unsigned int offset)
@@ -84,7 +82,6 @@
 			cpu_boot_rsp_set(CPU_MSG_OK);
 			res = FOUND_NO_RETURN;
 		}
-		preboot_disable_iatu_access();
 		break;
 	case KMD_MSG_FIT_RDY:
 		if (sts.is_fit_ready_ignored)
@@ -94,7 +91,6 @@
 		/* Tell LKD we got the message */
 		cpu_boot_rsp_set(CPU_MSG_OK);
 		preboot_puts("LKD FIT RDY received\n");
-		preboot_disable_iatu_access();
 		break;
 	case KMD_MSG_RST_DEV:
 		res = FOUND_NO_RETURN;
@@ -102,12 +98,10 @@
 		cpu_boot_cmd_clr_before_wfe();
 		cpu_boot_rsp_set(CPU_MSG_OK);
 		preboot_puts("LKD RST DEV received\n");
-		preboot_disable_iatu_access();
 		break;
 	case KMD_MSG_SKIP_BMC:
 		cpu_boot_rsp_set(CPU_MSG_ERR);
 		preboot_puts("LKD SKIP BMC received\n");
-		preboot_disable_iatu_access();
 		/* no break */
 	default:
 		res = NOT_FOUND;
@@ -285,7 +279,6 @@
 	case COMMS_GOTO_WFE:
 	case COMMS_SKIP_BMC:
 	case COMMS_LOW_PLL_OPP:
-		preboot_disable_iatu_access();
 		res = FOUND_FAST_RETURN;
 		break;
 	default:
@@ -315,8 +308,6 @@
 void module_lkd_run_fast(void)
 {
 	const char *cmd_str = NULL;
-
-	preboot_disable_iatu_access();
 
 	switch (sts.cmd.cmd) {
 	case COMMS_NOOP:
